[continuous-deploy-fingerprint] Add platform as input

NoZiL opened the pull request:
Playing with the new action for sdk 51, I've noticed that the platform is missing.
This is an attempt to handle this case.

brentvatne reviewed (approved):
can you update the readme as well?

wschurman commented on src/actions/continuous-deploy-fingerprint.ts:
```diff
@@ -30,63 +33,103 @@ export async function continuousDeployFingerprintAction(input = collectContinuou
 
-  info(`Android fingerprint: ${androidFingerprintHash}`);
-  info(`iOS fingerprint: ${iosFingerprintHash}`);
+  async function buildAndroid() {
```
```suggestion
  async function buildAndroidIfNecessaryAsync() {
```

wschurman commented on src/actions/continuous-deploy-fingerprint.ts:
```diff
@@ -30,63 +33,103 @@ export async function continuousDeployFingerprintAction(input = collectContinuou
-  info(`Android fingerprint: ${androidFingerprintHash}`);
-  info(`iOS fingerprint: ${iosFingerprintHash}`);
+  async function buildAndroid() {
+    if (platform === 'ios') return;
```
small nit (null is clearer in result type than void):
```suggestion
    if (platform === 'ios') {
      return null;
    }
```

wschurman commented on src/actions/continuous-deploy-fingerprint.ts:
```diff
@@ -30,63 +33,103 @@ export async function continuousDeployFingerprintAction(input = collectContinuou
-  } else {
-    info(`No existing iOS build found for fingerprint, starting a new build...`);
-    iosBuildInfo = await createEASBuildAsync({ cwd: input.workingDirectory, platform: 'ios', profile: input.profile });
+  async function buildIos() {
```
```suggestion
  async function buildIosIfNecessaryAsync() {
```

wschurman commented on src/actions/continuous-deploy-fingerprint.ts:
```diff
@@ -30,63 +33,103 @@ export async function continuousDeployFingerprintAction(input = collectContinuou
-    info(`No existing iOS build found for fingerprint, starting a new build...`);
-    iosBuildInfo = await createEASBuildAsync({ cwd: input.workingDirectory, platform: 'ios', profile: input.profile });
+  async function buildIos() {
+    if (platform === 'android') return;
```
```suggestion
    if (platform === 'android') {
      return null;
    }
```

wschurman commented on src/actions/continuous-deploy-fingerprint.ts:
```diff
@@ -30,63 +33,103 @@ export async function continuousDeployFingerprintAction(input = collectContinuou
   }
 
-  const builds = [androidBuildInfo, iosBuildInfo];
+  const [android, ios] = await Promise.all([buildAndroid(), buildIos()]);
```
```suggestion
  const [androidResult, iosResult] = await Promise.all([
    buildAndroidIfNecessaryAsync(), 
    buildIosIfNecessaryAsync(),
  ]);
```

wschurman commented on src/actions/continuous-deploy-fingerprint.ts:
```diff
@@ -30,63 +33,103 @@ export async function continuousDeployFingerprintAction(input = collectContinuou
+      config,
+      projectId,
+      updates,
+      builds: [android?.buildInfo, ios?.buildInfo].filter(e => e !== undefined) as BuildInfo[],
```
```suggestion
      builds: [android?.buildInfo, ios?.buildInfo].filter(e => e !== null) as BuildInfo[],
```

wschurman reviewed (changes requested):
Good addition! Just a few stylistic changes, then I'll merge!

NoZiL committed:
add platform

NoZiL commented:
Hey @brentvatne, I added the input in the readme, tell me if something else needs to be added.
Hey @wschurman, sorry for the delay, I've dealt with the conflicts and added your suggestions. I also added the strict null checks in tsconfig to lint accordingly.

wschurman commented:
Thanks for the contribution!
